babel/messages/frontend.py

```diff
@@ -893,9 +893,16 @@ def run(self, argv=None):
             for identifier in identifiers:
                 locale = Locale.parse(identifier)
                 output = format % (identifier, locale.english_name)
-                print(output.encode(sys.stdout.encoding or
-                                    getpreferredencoding() or
-                                    'ascii', 'replace'))
+                # Prevent errors on printing by doing an encode/decode
+                # roundtrip with errors='replace'.
+                encoding = (
+                    sys.stdout.encoding
+                    or getpreferredencoding()
+                    or 'ascii'
+                )
+                print(
+                    output.encode(encoding, errors='replace').decode(encoding)
+                )
```

Comment:
I think in these Py3-only days where we also have `PYTHONUTF8` for Windows, this could just be

```suggestion
                print(output)
```